com.unity.multiplayer.mlapi/Tests/Runtime/Profiling/NetworkVariableNameTests.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using MLAPI.NetworkVariable;
+using MLAPI.NetworkVariable.Collections;
+using NUnit.Framework;
+
+namespace MLAPI.RuntimeTests.Profiling
+{
+    public sealed class NetworkVariableNameTests
+    {
+        NetworkManager m_NetworkManager;
+        NetworkVariableNameComponent m_NetworkVariableNameComponent;
+
+        [SetUp]
+        public void SetUp()
+        {
+            NetworkManagerHelper.StartNetworkManager(out m_NetworkManager);
+
+            var gameObjectId = NetworkManagerHelper.AddGameNetworkObject(Guid.NewGuid().ToString());
+            m_NetworkVariableNameComponent= NetworkManagerHelper.AddComponentToObject<NetworkVariableNameComponent>(gameObjectId);
+            NetworkManagerHelper.SpawnNetworkObject(gameObjectId);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            NetworkManagerHelper.ShutdownNetworkManager();
+        }
+
+        [Test]
+        public void EnsureNetworkVariableNameInitialization()
+        {
+            Assert.True(m_NetworkVariableNameComponent.NetworkVarString.Name.Contains(nameof(NetworkVariableNameComponent.NetworkVarString)));
+            Assert.True(m_NetworkVariableNameComponent.NetworkVarSet.Name.Contains(nameof(NetworkVariableNameComponent.NetworkVarSet)));
+            Assert.True(m_NetworkVariableNameComponent.NetworkVarList.Name.Contains(nameof(NetworkVariableNameComponent.NetworkVarList)));
+            Assert.True(m_NetworkVariableNameComponent.NetworkVarDictionary.Name.Contains(nameof(NetworkVariableNameComponent.NetworkVarDictionary)));
```

Comment:
wouldn't these should be equal rather than contains?

Reply:
Unfortunately, depending on how you declare your variable (property vs field) it's going to be different.

In this specific example, the first variable name is <NetworkVarString>k__BackingField

For our purposes, we can just pretty-print it for display. Maybe we can change Name to something that sounds a bit less "safe", or add a comment explaining this.

Reply:
Changed it to StringAssert since it's cleaner